hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -927,6 +924,40 @@ private static File urlToFile(@Nonnull URL url) {
         return resourceConfigs;
     }
 
+    /**
+     * Registers the given serializer for the given class for the scope of the
+     * job. It will be accessible to all the code attached to the underlying
+     * pipeline or DAG, but not to any other code. (An important example is the
+     * {@code IMap} data source, which can instantiate only the classes from
+     * the Jet instance's classpath.)
+     * <p>
+     * Serializer must have no-arg constructor.
+     *
+     * @return {@code this} instance for fluent API
+     */
+    @Nonnull
+    @EvolvingApi
+    public <T, S extends StreamSerializer<?>> JobConfig registerStreamSerializer(@Nonnull Class<S> serializerClass,
```

Comment:
can we rename method to `registerSerializerFor` and have the parameters reversed. `JobConfig.registerSerializerFor(Value.class, ValueSerializer.class)`